pkg/cmd/agentversionjob.go

```diff
@@ -128,44 +59,51 @@ func (avj *AgentVersionCheckJob) Status() error {
 
 // Execute - run agent version check job one time
 func (avj *AgentVersionCheckJob) Execute() error {
-	err := avj.getJFrogVersions()
+	state, err := avj.getAgentState()
 	if err != nil {
 		log.Trace(err)
-		// Could not get update from jfrog.  Warn that we could not determine version and continue processing
-		log.Warn("Agent cannot determine the next available release. Be aware that your agent could be outdated.")
-	} else {
-		// Successfully got jfrog version.  Now compare build to latest version
-		if isVersionStringOlder(avj.buildVersion, config.AgentLatestVersion) {
-			log.Warnf("New version available. Please consider upgrading from version %s to version %s", avj.buildVersion, config.AgentLatestVersion)
-		}
+		// Could not get update from agent state.  Warn that we could not determine version and continue processing
+		log.Warn("Agent cannot determine the current available release. Be aware that your agent could be outdated.")
+	}
+
+	switch state {
+	case "available":
+		log.Warn("Please be aware that there is a newer agent version available")
+	case "outdated":
+		log.Error("current agent version is no longer supported. We strongly advise to update the agent as soon as possible.")
+	case "retracted":
+		log.Error("current agent version has a known issue, please update the agent immediately.")
 	}
 	return nil
 }
 
-// getJFrogVersions - obtaining the versions from JFrog website
-// **Note** polling the jfrog website is the current solution to obtaining the list of versions
-// In the future, adding a (Generic) resource for grouping versions together under the same scope is a possible solution
-// ie: a new unscoped resource that represents the platform services, so that other products can plug in their releases.
-func (avj *AgentVersionCheckJob) getJFrogVersions() error {
-	request := coreapi.Request{
-		Method:  http.MethodPost,
-		URL:     jfrogURL,
-		Headers: avj.headers,
-		Body:    avj.requestBytes,
+func (avj *AgentVersionCheckJob) getAgentState() (string, error) {
+	agentRes := avj.manager.GetAgentResource()
+	if agentRes == nil {
+		return "", fmt.Errorf("could not get the agent resource")
 	}
-	response, err := avj.apiClient.Send(request)
-	if err != nil {
-		return err
+
+	// The kind should be only DA or TA
+	subResKey := management.DiscoveryAgentAgentstateSubResourceName
+	if agentRes.GetGroupVersionKind().Kind == "TraceabilityAgent" {
+		subResKey = management.DiscoveryAgentAgentstateSubResourceName
```

Comment:
Although it the same value but should be TraceabilityAgentAgentstateSubResourceName